overlord/snapstate/aliases.go

```diff
@@ -132,27 +164,38 @@ func (m *SnapManager) doAlias(t *state.Task, _ *tomb.Tomb) error {
 		aliasStatuses = make(map[string]string)
 	}
 	var add []*backend.Alias
-	for alias := range toEnable {
+	var remove []*backend.Alias
+	for alias, newStatus := range toggles {
 		aliasApp := curInfo.Aliases[alias]
 		if aliasApp == nil {
-			return fmt.Errorf("cannot enable alias %q for %q, no such alias", alias, snapName)
+			return fmt.Errorf("cannot toggle alias %q for %q, no such alias", alias, snapName)
 		}
-		if aliasStatuses[alias] == "enabled" {
+		if aliasStatuses[alias] == newStatus {
 			// nothing to do
 			continue
 		}
-		err := checkAliasConflict(st, snapName, alias)
-		if err != nil {
-			return err
+		switch newStatus {
+		case "enabled":
+			err := checkAliasConflict(st, snapName, alias)
+			if err != nil {
+				return err
+			}
+			add = append(add, &backend.Alias{
+				Name:   alias,
+				Target: filepath.Base(aliasApp.WrapperPath()),
+			})
+		case "disabled":
+			if aliasStatuses[alias] != "" {
+				remove = append(remove, &backend.Alias{
```

Comment:
The appended value is the same in both cases, so it can be moved out of the switch.